src/oracles/reserve/lrt/WeEthWethReserveOracle.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import { ReserveOracle } from "../ReserveOracle.sol";
+import { BASE_WEETH_ETH_EXCHANGE_RATE_CHAINLINK } from "../../../Constants.sol";
+import { WadRayMath } from "../../../libraries/math/WadRayMath.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/**
+ * @notice Reserve oracle for weETH.
+ *
+ * @custom:security-contact security@molecularlabs.io
+ */
+contract WeEthWethReserveOracle is ReserveOracle {
+    using WadRayMath for uint256;
+    using SafeCast for int256;
+
+    error MaxTimeFromLastUpdateExceeded(uint256, uint256);
+
+    uint256 public immutable MAX_TIME_FROM_LAST_UPDATE; // seconds
+
+    /**
+     * @notice Creates a new `WeEthWethReserveOracle` instance. Provides
+     * the amount of WETH equal to one weETH.
+     * ETH / eETH is 1 since eETH is rebasing. Depeg here would reflect in eETH / wETH
+     * exchange rate.
+     * @dev The value of weETH denominated in WETH by the provider.
+     * @param _ilkIndex of weETH.
+     * @param _feeds List of alternative data sources for the weETH exchange rate.
+     * @param _quorum The amount of alternative data sources to aggregate.
+     * @param _maxChange Maximum percent change between exchange rate updates. [RAY]
+     */
+    constructor(
+        uint8 _ilkIndex,
+        address[] memory _feeds,
+        uint8 _quorum,
+        uint256 _maxChange,
+        uint256 _maxTimeFromLastUpdate
+    )
+        ReserveOracle(_ilkIndex, _feeds, _quorum, _maxChange)
+    {
+        MAX_TIME_FROM_LAST_UPDATE = _maxTimeFromLastUpdate;
+        _initializeExchangeRate();
+    }
+
+    /**
+     * @notice Returns the exchange rate between WETH and weETH.
+     * @return Exchange rate between WETH and weETH.
+     */
+    function _getProtocolExchangeRate() internal view override returns (uint256) {
+        (, int256 ethPerWeEth,, uint256 ethPerWeEthUpdatedAt,) =
```

Comment:
might want to add in check on sequencerUptimeFeed and check also a grace period since it is back up per L2 chainlink specs (mentioned in tg)